### DISCUSSION TITLE
NestJS + Angular Example

### DISCUSSION THREAD

**Discussion by tobyStaff** [vercel/community#127] [Show & Tell]
Would be fantastic if you could provide an example / recipe / guide for building and deploying a NestJS API with Angular frontend.  There aren't many examples online of how to do it correctly and the examples we do have are a little fragmented at the moment. 

If you could include example of usage with connection to cloud db like MongoDB Atlas - that would be incredible. 

Thanks!

**Comment by mcsdevv** [comment]
Hello,

We have no plans to do this as NestJS typically relies on a running server and is not the best fit for Serverless Functions.

We recently added a guide on using Express (although absolutely not required) as many people will be more familiar with that. Many of the concepts will apply to other Node.js frameworks: https://vercel.com/guides/using-express-with-vercel

You can find how to manage MongoDB connections here: https://vercel.com/guides/deploying-a-mongodb-powered-api-with-node-and-vercel

**Reply by jdgamble555** [reply]
I keep hearing this excuse for why Vercel does not provide NestJS or Angular Universal.  This is clearly wrong.  If you can easily deploy Angular Universal or NestJS to Google Cloud functions or Google Cloud Run (which ARE serverless function), there is no reason to NOT provide this functionality for Vercel.  Please add this feature, as you are losing money directly by not providing this.

J

**Reply by Pieparker** [reply]
@jdgamble555 feel free to create a [feature suggestion](https://github.com/vercel/community/discussions/categories/feature-suggestions) capturing the specific requirements to make this work!

**Comment by tobyStaff** [comment]
Cheers Michael,

I think i'll opt for Heroku for the API & vercel for the front-ends. As it
seems I'm swimming against the tide on Vercel.

Appreciate your time.

Toby

**Reply by peraltafederico** [reply]
Same here!
